hazelcast/src/main/java/com/hazelcast/sql/impl/exec/fetch/FetchExec.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec.fetch;
+
+import com.hazelcast.sql.impl.exec.AbstractUpstreamAwareExec;
+import com.hazelcast.sql.impl.exec.Exec;
+import com.hazelcast.sql.impl.exec.IterationResult;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.row.RowBatch;
+import com.hazelcast.sql.impl.worker.QueryFragmentContext;
+
+/**
+ * An Executor that applies offset and limit clauses to the rows..
+ */
+public class FetchExec extends AbstractUpstreamAwareExec {
```

Comment:
As discussed in Slack, the `offset` would produce arbitrary results, because we cannot order rows. We need a motivated decision whether to release the `offset` support without sorting or not. 
Regarding the `fetch` alone without sorting, the alternative is to define the limit programmatically. We will have to add it anyway for JDBC `Statement.setMaxRows`.

Reply:
We disable FETCH/OFFSET in the nested SELECTs. See https://github.com/hazelcast/hazelcast/pull/18091/commits/baa0f80f059bded1eaf14e31798ce4d084f5604f